### PR TITLE
Feature/product details screen

### PR DESCRIPTION
**Pull request by ababicheva**

### CODE REVIEW

File: lib/presentation/product_details_route.dart [hunk 1]
```diff
@@ -0,0 +1,213 @@
+import 'package:flutter/material.dart';
+import 'package:flutterapp/messages.dart';
+
+/// Screen represents static Product Details [see in Sketch](https://www.sketch.com/s/a67f7ba8-f681-4ce8-abf4-ccb72cb57eb0/a/gx178z)
```
Comment (yakivmospan) [inline lib/presentation/product_details_route.dart]: Not static anymore? Now using `StatefulWidget`

File: lib/presentation/product_details_route.dart [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+              onPressed: () {
+                setState(() {
+                  _onFavoritePressed();
+                });
+              },
+            )
+          ],
+          elevation: 0,
+        ),
+        body: Center(
```
Comment (yakivmospan) [inline lib/presentation/product_details_route.dart]: Can you please move `body` widget to separate private `_ProductDetailsRouteContent` widget? Or not private, `ProductDetailsRouteContent` up to you

**Review by vyatsykiv — APPROVED** [review]

File: lib/presentation/common/widget/checkbox_list_widget.dart [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:flutterapp/presentation/common/viewdata/checkbox_list_view_data.dart';
+import 'package:flutterapp/widgets/ripple_widget.dart';
+
+class CheckboxListWidget extends StatefulWidget {
```
Comment (yakivmospan) [inline lib/presentation/common/widget/checkbox_list_widget.dart]: Why is it named as a `List` ?

File: lib/presentation/productDetails/product_details_route.dart [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+                _onFavoritePressed();
+              });
+            },
+          ),
+          /// TODO: show/hide this button depending on flag in bloc when it's implemented
+          IconButton(icon: SvgPicture.asset(AppImages.iconBasket))
+        ],
+        elevation: 0,
+      ),
+      body: SingleChildScrollView(
```
Comment (yakivmospan) [inline lib/presentation/productDetails/product_details_route.dart:214]: Can we please move the body to another widget, will simplify reading?

File: lib/presentation/common/widget/horizontal_radio_group_widget.dart [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> children = List();
+    for (int i = 0; i < items.length; i++) {
+      children.add(Flexible(
+        flex: 1,
+        child: GestureDetector(
+          child: Container(
+            alignment: Alignment.center,
+            padding: EdgeInsets.only(top: 9, bottom: 7),
```
Comment (yakivmospan) [inline lib/presentation/common/widget/horizontal_radio_group_widget.dart]: Should this be hardcoded?

File: lib/presentation/common/widget/radio_list_widget.dart [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      _checkedIndex = index;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> children = List();
+    for (int i = 0; i < items.length; i++) {
+      children.add(RippleWidget(
+          child: Column(
```
Comment (yakivmospan) [inline lib/presentation/common/widget/radio_list_widget.dart]: Can we please move this child to separate widget?

File: lib/presentation/common/widget/radio_list_widget.dart [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+            children: <Widget>[
+              Padding(
+                padding: const EdgeInsets.only(top: 5),
```
Comment (yakivmospan) [inline lib/presentation/common/widget/radio_list_widget.dart]: Should be hardcoded ?

File: lib/presentation/common/viewdata/checkbox_list_view_data.dart [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import 'package:flutter/material.dart';
+
+class CheckboxListViewData {
```
Comment (yakivmospan) [inline lib/presentation/common/viewdata/checkbox_list_view_data.dart]: Why is it `CheckboxListViewData`, can be just `CheckboxViewData` ?

File: lib/presentation/common/widget/checkbox_list_widget.dart (hunk 3 above)
Comment (yakivmospan) [reply lib/presentation/common/widget/checkbox_list_widget.dart]: Can be just `CheckboxWidget` ?

File: lib/presentation/common/widget/horizontal_radio_group_widget.dart [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> children = List();
+    for (int i = 0; i < items.length; i++) {
+      children.add(Flexible(
+        flex: 1,
+        child: GestureDetector(
```
Comment (yakivmospan) [inline lib/presentation/common/widget/horizontal_radio_group_widget.dart:55]: Why do we need GestureDetector ? onTap is from it ?

File: lib/presentation/common/widget/horizontal_radio_group_widget.dart [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+    setState(() {
+      _checkedIndex = index;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> children = List();
+    for (int i = 0; i < items.length; i++) {
+      children.add(Flexible(
```
Comment (yakivmospan) [inline lib/presentation/common/widget/horizontal_radio_group_widget.dart:53]: Can we separate child widgets a bit more?

File: lib/presentation/common/widget/horizontal_radio_group_widget.dart (hunk 9 above)
Comment (ababicheva) [reply lib/presentation/common/widget/horizontal_radio_group_widget.dart:55]: yes, to detect onTap

**Review by yakivmospan — APPROVED** [review]

**Pull request merged**